src/emqx_cm.erl

```diff
@@ -493,3 +522,23 @@ get_chann_conn_mod(ClientId, ChanPid) when node(ChanPid) == node() ->
 get_chann_conn_mod(ClientId, ChanPid) ->
     rpc_call(node(ChanPid), get_chann_conn_mod, [ClientId, ChanPid], ?T_GET_INFO).
 
+increment_connected_client_count() ->
+    ?tp(emqx_cm_connected_client_count_inc, #{}),
+    CRef = persistent_term:get({?MODULE, ?CONN_CLIENT_CTR}),
+    ok = counters:add(CRef, ?CONN_CLIENT_CTR_IDX, 1).
+
+decrement_connected_client_count() ->
+    ?tp(emqx_cm_connected_client_count_dec, #{}),
+    CRef = persistent_term:get({?MODULE, ?CONN_CLIENT_CTR}),
+    ok = counters:sub(CRef, ?CONN_CLIENT_CTR_IDX, 1).
+
+get_connected_client_count() ->
+    CRef = persistent_term:get({?MODULE, ?CONN_CLIENT_CTR}),
+    %% check if inconsistent; if so, reset to 0
+    case counters:get(CRef, ?CONN_CLIENT_CTR_IDX) of
+        N when N < 0 ->
+            counters:put(CRef, ?CONN_CLIENT_CTR_IDX, 0),
```

Comment:
This may introduce some race conditions? I.e. we may lose increments done after `get` and before `put`?
We may also try `ets:update_counter/4` version https://www.erlang.org/doc/man/ets.html#update_counter-4 that allows to decrement consistently, i.e. not to decrease below 0.

Reply:
I guess that ETS is going to be easier to make consistent/atomic, indeed.